drivers/usb/device/usb_dc_it82xx2.c

```diff
@@ -695,45 +676,151 @@ static inline void it82xx2_handler_out(uint8_t fifo_idx)
 		udata0.ep_data[ep_idx].cb_out(ep_idx, USB_DC_EP_DATA_OUT);
 	}
 
-	if (fifo_idx == 0) {
+	if (ep_idx == 0) {
 		/* SETUP_TOKEN follow OUT_TOKEN */
 		if (it82xx2_check_setup_following_out()) {
 			udata0.last_token = udata0.now_token;
 			udata0.now_token = SETUP_TOKEN;
 			udata0.st_state = SETUP_ST;
-			ep_regs[fifo_idx].ep_ctrl.fields.outdata_sequence_bit = 1;
+			ep_regs[ep_idx].ep_ctrl.fields.outdata_sequence_bit = 1;
 			udata0.ep_data[ep_idx].cb_out(ep_idx | USB_EP_DIR_OUT, USB_DC_EP_SETUP);
 
 			if (udata0.no_data_ctrl) {
-				ep_regs[fifo_idx].ep_ctrl.fields.ready_bit = 1;
+				ep_regs[ep_idx].ep_ctrl.fields.ready_bit = 1;
 				udata0.no_data_ctrl = false;
 			}
 		}
 	}
 }
 
+static bool get_extend_enable_bit(const uint8_t ep_idx)
+{
+	union epn_extend_ctrl1_reg *epn_ext_ctrl1 = NULL;
+	bool enable;
+
+	epn_ext_ctrl1 = (union epn_extend_ctrl1_reg *)it82xx2_get_ext_ctrl(ep_idx, EP_ENABLE);
+	if (((ep_idx - 4) / 3 == 0)) {
+		enable = !!(epn_ext_ctrl1->fields.epn0_enable_bit);
+	} else if (((ep_idx - 4) / 3 == 1)) {
+		enable = !!(epn_ext_ctrl1->fields.epn3_enable_bit);
+	} else if (((ep_idx - 4) / 3 == 2)) {
+		enable = !!(epn_ext_ctrl1->fields.epn6_enable_bit);
+	} else {
+		enable = !!(epn_ext_ctrl1->fields.epn9_enable_bit);
+	}
+	return enable;
+}
+
+static bool get_extend_ready_bit(const uint8_t ep_idx)
+{
+	struct usb_it82xx2_regs *const usb_regs = (struct usb_it82xx2_regs *)it82xx2_get_usb_regs();
+	struct epn_ext_ctrl_regs *ext_ctrl =
+		usb_regs->fifo_regs[EP_EXT_REGS_DX].ext_0_3.epn_ext_ctrl;
+	int idx = ((ep_idx - 4) % 3) + 1;
+
+	return !!(ext_ctrl[idx].epn_ext_ctrl2 & BIT((ep_idx - 4) / 3));
```

Comment:
Conversion to bool type is described in e.g. C99 spec. Why is there `!!`?

Reply:
Fixed. Thanks.